src/lib/jsonRpc.ts

```diff
@@ -0,0 +1,60 @@
+import { DEFAULT_LOCAL_RPC_ADDRESS } from "./consts";
+import type { Rpc, SignedRpc } from "./dealProposal";
+
+type Either<L, R> = L | R;
+type RpcResult<T> = Either<T, Error>;
+
+// Helper function for JSON-RPC calls
+async function callJsonRpc<R>(url: string, params: object[], method: string): Promise<RpcResult<R>> {
```

Comment:
https://www.npmjs.com/package/@polkadot/rpc-provider?
or maybe `papi` (https://papi.how/providers/ws#ws-provider) has something? Feels like we shouldn't be writing and maintaning this boilerplate.